pkg/api/alerting.go

```diff
@@ -693,7 +693,15 @@ func (hs *HTTPServer) PauseAlert(legacyAlertingEnabled *bool) func(c *models.Req
 			return response.Error(500, "Get Alert failed", err)
 		}
 
-		guardian := guardian.New(c.Req.Context(), query.Result.DashboardId, c.OrgID, c.SignedInUser)
+		dashUID, err := hs.getDashboardUID(c.Req.Context(), query.Result.DashboardId, c.OrgID)
+		if err != nil {
+			return response.Error(500, "Get dashboard failed", err)
+		}
+
+		guardian, err := guardian.New(c.Req.Context(), dashUID, c.OrgID, c.SignedInUser)
+		if err != nil {
+			return response.ErrOrFallback(http.StatusInternalServerError, "Error while creating permission guardian", err)
+		}
```

Comment:
As is this will add additional db queries to check permissions on a dashboard. First translate id to uid, then in guardian trasnalte the uid to id. Seems like we should add additional constructors for dashboard guardian e.g. NewFromID(), NewFromUID() and maybe one for when we aleardy have the full dashboard model available (to skip additional query to get model)